job_runner/main.py

```
@@ -17,40 +16,129 @@
+) -> bool:
+    if file_changes:
+        return False
+
+    return _looks_like_edit_request(prompt) or _claims_file_changes(response_text)
+
 # Redis client for cloud run
 redis_client:redis.Redis = None
 
 async def init_redis():
     """Initialize Redis connection with SSL support"""
     global redis_client
```

Ruff (0.15.10)

[warning] Using the global statement to update `redis_client` is discouraged

(PLW0603)